### PR TITLE
Update dependent modules

### PR DESCRIPTION
**Pull request by kuenishi**
- [x] Update node_package, getopt, folsom, cluster_info to the latest
- [x] Update riak-erlang-client and riak_repl_pb_api to 2.0.1/2.0.0
- [x] lager 2.1.0 conflicts with lager_syslog 2.0.3 => not using
- [x] riak_cs_multibag will be pinned with rebar_lock_deps_plugin
- [x] Webmachine will be updated to upcoming 1.10.7
- [x] update webmachine (https://github.com/basho/webmachine/pull/233) for https://github.com/basho/mochiweb/pull/17

### CODE REVIEW

**Comment by kuenishi** [comment]
Fmm. Still having errors like

```
==> riak_cs (dialyzer)
riak_cs_block_server.erl:318: The pattern Success = {'ok', _} can never match the type {'error',_}
riak_cs_block_server.erl:353: The pattern {'ok', RiakObject} can never match the type {'error',_}
riak_cs_config.erl:239: The pattern {'ok', ClusterID} can never match the type {'error',_}
tools.mk:38: recipe for target 'dialyzer-run' failed
make: *** [dialyzer-run] Error 1
```

Looks like hard part of riak_pb is related.

**Comment by shino** [comment]
> riak_cs_multibag will be pinned with rebar_lock_deps_plugin

IIRC, the plugin uses hardcoded locked config file. This repository uses an environment variable to build two flavors (oss and enterprise). I'm wondering how these work well together :thought_balloon:

**Comment by shino** [comment]
Just for a memo: one issue I know now about webmachine/mochiweb upgrading.
There is a little discrepancy how to treat range header in corner case `bytes=0-`,
which can be treated as partial contents as well as normal non-range contents.

The changes in webmachine and forked mochiweb were [1] and [2].

[1] https://github.com/basho/mochiweb/pull/7
[2] https://github.com/basho/webmachine/pull/124

**Comment by kuenishi** [comment]
I decided to ignore those three errors because they all come from riak_pb types and don't want to dive deep there. As Shino mentioned, `object_get_test` and `external_client_tests` are now failing.

**Comment by kuenishi** [comment]
basho/webmachine#124 is included in the latest webmachine develop (which is to be tagged as 1.10.7). basho/mochiweb#7 is waiting for merge as basho/mochiweb#17 .

**Comment by shino** [comment]
`upgrade_downgrade_test` fails with multibag configuration.

```
================ upgrade_downgrade_test failure stack trace =====================
{{aws_error,{http_error,404,[],
                        <<"<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Resource>/riak-test-bucket-foobar/riak_test_key1</Resource><RequestId></RequestId></Error>">>}},
 [{erlcloud_s3,s3_request,9,[{file,"src/erlcloud_s3.erl"},{line,898}]},
  {erlcloud_s3,fetch_object,5,[{file,"src/erlcloud_s3.erl"},{line,456}]},
  {upgrade_downgrade_test,basic_get_test_case,4,
                          [{file,"riak_test/tests/upgrade_downgrade_test.erl"},
                           {line,147}]},
  {upgrade_downgrade_test,verify_all_data,2,
                          [{file,"riak_test/tests/upgrade_downgrade_test.erl"},
                           {line,135}]},
  {upgrade_downgrade_test,confirm,0,
                          [{file,"riak_test/tests/upgrade_downgrade_test.erl"},
                           {line,64}]},
  {riak_test_runner,return_to_exit,3,
                    [{file,"src/riak_test_runner.erl"},{line,159}]}]}
=================================================================================
```

**Comment by kuenishi** [comment]
I think we should merge https://github.com/basho/riak_cs_multibag/pull/16 first.

**Comment by kuenishi** [comment]
@shino fmmm. It worked for me with the latest multibag.

**Comment by kuenishi** [comment]
And https://github.com/basho/webmachine/pull/233 followed by 1.10.7 tag will be the last dependent work.

**Comment by kuenishi** [comment]
Will run r_t on Monday. riak_cs_multibag will be tagged just before packaging/performance testing/release.

**Comment by kuenishi** [comment]
@shino I reproduced! My rt configuration was bad; `previous` CS was not the EE one.

**Comment by kuenishi** [comment]
Fixed

File: src/riak_cs_block_server.erl [hunk 1]
```diff
@@ -270,6 +271,9 @@ do_get_block(ReplyPid, Bucket, Key, ClusterID, UseProxyGet, ProxyActive,
                   Timeout, ProceedFun, RetryFun, NumRetries, UseProxyGet,
                   ProxyActive, ClusterID).
 
+-spec try_local_get(riak_client(), binary(), binary(), list(), list(),
+                    timeout(), fun(), fun(), non_neg_integer(), boolean(),
+                    boolean(), binary()) -> no_return().
```
Comment (shino) [inline src/riak_cs_block_server.erl]: The doc[1] says `no_return()` = `none()` is bottom type. But this function can return `ok` or any term that `ProceedFun` returns. It should be `any()` or `term()`. 

[1] http://www.erlang.org/doc/reference_manual/typespec.html

File: src/riak_cs_block_server.erl [hunk 2]
```diff
@@ -291,6 +295,9 @@ try_local_get(RcPid, FullBucket, FullKey, GetOptions1, GetOptions2,
             RetryFun(failure)
     end.
 
+-spec handle_local_notfound(riak_client(), binary(), binary(), list(),
+                            fun(({ok,binary()}) -> no_return()), fun((non_neg_integer())->no_return()), non_neg_integer(), boolean(),
+                            boolean(), binary()) -> no_return().
```
Comment (shino) [inline src/riak_cs_block_server.erl]: Same as above.

File: src/riak_cs_block_server.erl [hunk 3]
```diff
@@ -291,6 +295,9 @@ try_local_get(RcPid, FullBucket, FullKey, GetOptions1, GetOptions2,
             RetryFun(failure)
     end.
 
+-spec handle_local_notfound(riak_client(), binary(), binary(), list(),
+                            fun(({ok,binary()}) -> no_return()), fun((non_neg_integer())->no_return()), non_neg_integer(), boolean(),
```
Comment (shino) [inline src/riak_cs_block_server.erl]: I guess same for these two `no_return()`.

**Comment by kuenishi** [comment]
Fixed the spec and force-pushed :smiling_imp:

**Comment by kuenishi** [comment]
@borshop merge

**Pull request merged**